spec/index.adoc

```diff
@@ -1674,6 +1674,101 @@ Any user can top-up any canister this way.
 
 This method is only available in local development instances.
 
+
+[#ic-bitcoin-canisters]
+== The IC bitcoin canisters
+
+Similar to the <<ic-management-canister,management canister>>, the Bitcoin functionality is also exposed through a facade, which we refer to as the "Bitcoin canisters".
+There are two Bitcoin canisters differing only in that one operates on the real Bitcoin blockchain whereas the other only works with the Bitcoin testnet.
+The Bitcoin canisters can be accessed using their addresses:
+
+- The canister address of the Bitcoin canister that uses the real Bitcoin blockchain is `u25tu-yqaaa-aaaaa-aaaaa-eai`.
+- The canister address of the Bitcoin canister that uses the Bitcoin testnet is `jf4vc-xabaa-aaaaa-aaaaa-eai`.
+
+Invoking functions on these canisters will cost cycles. The concrete costs for each function are yet to be determined.
+
+Information about Bitcoin can be found in the https://developer.bitcoin.org/devguide/[Bitcoin developer guides].
+
+[#ic-bitcoin-candid]
+=== Interface overview
+
+The following interface description, in https://github.com/dfinity/candid/blob/master/spec/Candid.md[Candid syntax], describes the available functionality. You can also link:{attachmentsdir}/bitcoin.did[download the file].
+----
+include::{example}bitcoin.did[]
+----
+
+The binary encoding of arguments and results are as per Candid specification.
+
+[#ic-bitcoin_get_utxos]
+=== IC method `get_utxos`
+
+Given a https://en.bitcoin.it/wiki/Base58Check_encoding[base58-encoded] address as part of the
+`GetUtxosRequest` , the function returns all unspent transaction outputs (UTXOs) associated with the
+provided address based on the current view of the Bitcoin blockchain available to the Bitcoin canister.
+
+If the call fails, e.g., because the address is malformed, a `GetUtxosError` variant is returned,
+indicating the reason for the failed call.
+
+The optional `filter` parameter can be used to restrict the set of returned UTXOs, either providing a
+minimum number of confirmations or a pagination record.
+In the first case, only UTXOs with at least the provided number of confirmations
+are returned, i.e., transactions with fewer than this number of
+confirmations are not considered. In other words, if the number of confirmations is `c`, an output
+is returned if it occurred in a transaction with at least `c` confirmations
+and there is no transaction that spends the same output with at least `c` confirmations.
+In the second case, a pagination record must be provided, specifying a block hash that constitutes
+the tip of the considered chain for this request and, optionally, an offset and a limit.
```

Comment:
What about the case where this is no `filter` present? Would it make sense to explicitly address that?

Reply:
See line 1727ff. Is this information sufficient?
Would you prefer to have that paragraph first and then talk about the workflow?

Reply:
Ah, I didn't notice it further down below. I think it'd be a bit more clear to first talk about all the possible cases, and then talk about the recommended workflow, but that's not a strong preference.

Reply:
I agree. I changed the order.